### PR TITLE
Ports: Add SerenityOS theming repository code as ported software

### PR DESCRIPTION
**Pull request by supercomputer7**
This makes it easier to use the external themes in the SerenityOS/theming repository. We might want to use the master branch instead of sticking this to a commit hash.
cc @Xexxa @djwisdom

### CODE REVIEW

**Comment by djwisdom** [comment]
This is a straight-forward solution, very good even :^)

**Comment by EWouters** [comment]
It would indeed be much nicer to use the master branch like we do with other projects that live in the SerenityOS org. There are some security considerations though.

**Comment by supercomputer7** [comment]
> It would indeed be much nicer to use the master branch like we do with other projects that live in the SerenityOS org. There are some security considerations though.

I rather keep it to a commit hash because of the security considerations that we insist on them for other ports. I'm sure that if we keep updating it in a sane schedule, then it's not that a bad thing. Although I trust @xexxa and @djwisdom to maintain that repo in an excellent way, mistakes do happen and using the master branch is certainly not a good idea for any ported software.

Also, I'd suggest to use tags in the theming repository (preferably with some sort of version cycle), as it will make it easier to choose a version, and not using an arbitrary commit hash. I suggest this because I found it quite annoying to see that one day after I updated my linux setup, the Arc theme for my XFCE installation changed making the look and feel much more worse in my opinion, without the proper way to revert those changes... so I take this example of something I'd like to prevent from happening in Serenity.

**Comment by Xexxa** [comment]
> > It would indeed be much nicer to use the master branch like we do with other projects that live in the SerenityOS org. There are some security considerations though.
> 
> I rather keep it to a commit hash because of the security considerations that we insist on them for other ports. I'm sure that if we keep updating it in a sane schedule, then it's not that a bad thing. Although I trust @Xexxa and @djwisdom to maintain that repo in an excellent way, mistakes do happen and using the master branch is certainly not a good idea for any ported software.
> 
> Also, I'd suggest to use tags in the theming repository (preferably with some sort of version cycle), as it will make it easier to choose a version, and not using an arbitrary commit hash. I suggest this because I found it quite annoying to see that one day after I updated my linux setup, the Arc theme for my XFCE installation changed making the look and feel much more worse in my opinion, without the proper way to revert those changes... so I take this example of something I'd like to prevent from happening in Serenity.

We will make mistakes like everyone else, the fewer attack vectors, the better. Using the commit hash make sure there is an extra manual review before arbitrary data is allowed to be downloaded with the port script.

Not sure about the tags, feels weird when /theming holding so many different assets from different authors to group them together in releases, that would also afaik hold back all updates for all assets just to preserve a certain version of a theme. Wouldn't it be easier just to go to the commit history if you don't like a change and make a copy of the font/theme/cursor set/etc?

**Review by Xexxa — APPROVED** [review]
I like the concept of this port. It also works just like advertised.

**Comment by supercomputer7** [comment]
> > > It would indeed be much nicer to use the master branch like we do with other projects that live in the SerenityOS org. There are some security considerations though.
> > 
> > 
> > I rather keep it to a commit hash because of the security considerations that we insist on them for other ports. I'm sure that if we keep updating it in a sane schedule, then it's not that a bad thing. Although I trust @Xexxa and @djwisdom to maintain that repo in an excellent way, mistakes do happen and using the master branch is certainly not a good idea for any ported software.
> > Also, I'd suggest to use tags in the theming repository (preferably with some sort of version cycle), as it will make it easier to choose a version, and not using an arbitrary commit hash. I suggest this because I found it quite annoying to see that one day after I updated my linux setup, the Arc theme for my XFCE installation changed making the look and feel much more worse in my opinion, without the proper way to revert those changes... so I take this example of something I'd like to prevent from happening in Serenity.
> 
> We will make mistakes like everyone else, the fewer attack vectors, the better. Using the commit hash make sure there is an extra manual review before arbitrary data is allowed to be downloaded with the port script.
> 
> Not sure about the tags, feels weird when /theming holding so many different assets from different authors to group them together in releases, that would also afaik hold back all updates for all assets just to preserve a certain version of a theme. Wouldn't it be easier just to go to the commit history if you don't like a change and make a copy of the font/theme/cursor set/etc?

So there it is, we will keep it sticking to a commit hash. And yes, I could probably just clone the repo and take what I need, versioning makes this a bit easier in my opinion. Nevertheless, let's get this merged.

**Comment by Xexxa** [comment]
Draft icon for this port :^)

animated

(click for animation)

File: Ports/serenity-themes/package.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=serenity-themes
```
Comment (linusg) [inline Ports/serenity-themes/package.sh:2]: Let's keep the `theming` name, fonts are not themes :)

**Review by linusg — COMMENTED** [review]
Small nitpick, rest LGTM. Thanks for doing this!

**Pull request merged**